WooCommerce: Setup/NUX: Add manage egg UX

allendav opened the pull request:
See #13612 and #13466 

To test:
- navigate to /store/:site
- ensure the dashboard appears
- ensure the 'Add a Product' button takes you /store/products/:site/add
- ensure the 'Set up shipping' button takes you to /store/settings/:site/shipping
- ensure the 'Set up payments' button takes you to /store/settings/:site/payments
- ensure the 'Set up taxes' button takes you to /store/settings/:site/tax
- ensure the 'Customize' button takes you to the customizer in wp-admin on your site

Note: The following is _NOT_ working in this PR and will be handled later/next in a separate PR:
- documentation links all point to https://support.wordpress.com/ right now since docs are not available at this time
- the footer buttons don't do anything yet, because we don't have Fledgling or Momma views of the Dashboard available yet
- the checkboxes are hard coded to all off (except customize) because we do not have those bits in state yet

cc @jameskoster for design love - I think I imported the checklist lady SVG from the sketch file improperly - the colors don't look quite right

jameskoster commented:
Hoping to get to this later today. A couple of content tweaks you for you;

![egg](https://cldup.com/EzfE4rrHWD-3000x3000.png)

* Remove the "I'll do this later" link
* Add "I won't be shipping" link
* Add "I'm not charging sales tax" link

allendav commented:
@jameskoster - "I'll do this later link" removed in https://github.com/Automattic/wp-calypso/pull/14202/commits/a1076870ba212699f171e7b94f0e566f12b2ddcc

allendav commented:
@jameskoster - secondary actions (and hiding) added for shipping and for taxes in https://github.com/Automattic/wp-calypso/pull/14202/commits/84abbe2437f895f60119ec3ed572852eae492df9

allendav commented:
@justinshreve @belcherj - all rebased and ready to review

jameskoster commented:
Argh, I'm getting an empty screen on this branch now. No errors. Going to work on the illustration in the mean time.

justinshreve commented on client/extensions/woocommerce/app/dashboard/setup-header.js:
```diff
@@ -0,0 +1,23 @@
+		<div className="dashboard__setup-header">
+			{ imageSource ? <img src={ imageSource } width={ imageWidth } className="dashboard__setup-header-image" /> : null }
+			{ <h2 className="dashboard__setup-header-title">{ title }</h2> }
+			{ subtitle ? <h3 className="dashboard__setup-header-subtitle">{ subtitle }</h3> : null }
```
If we use the syntax I've come to favor so much, we could drop the ternary and `null` in markup for checks like this:

`{ subtitle && ( <h3 className="dashboard__setup-header-subtitle">{ subtitle }</h3> ) }`

allendav replied:
LOL :) Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

justinshreve commented on client/extensions/woocommerce/app/dashboard/setup-task.js:
```diff
@@ -0,0 +1,89 @@
+class SetupTask extends Component {
+	static propTypes = {
+		actions: PropTypes.arrayOf( PropTypes.shape( {
+			isSecondary: PropTypes.string.bool,
```
Typo: `PropTypes.bool`.

allendav replied:
Oy! Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

justinshreve commented on client/extensions/woocommerce/app/dashboard/setup-header.js:
```diff
@@ -0,0 +1,23 @@
+};
+
+SetupHeader.propTypes = {
+	imageSource: React.PropTypes.string,
```
Let's destructure `PropTypes` in the import above.

allendav replied:
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

justinshreve commented on client/extensions/woocommerce/app/dashboard/setup.js:
```diff
@@ -0,0 +1,196 @@
+class Setup extends Component {
+	static propTypes = {
+		onFinished: PropTypes.func,
+		site: React.PropTypes.shape( {
```
Both this line and the `slug` line below can just use `PropTypes` instead of `React.PropTypes`.

allendav replied:
Agreed! Copypasta. Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/032e34b77886a58b7e5172370e876738a9c9176d

justinshreve commented on client/extensions/woocommerce/app/dashboard/setup.js:
```diff
@@ -0,0 +1,196 @@
+					subtitle={ translate( 'Below you will find the essential tasks to complete before making your store live.' ) }
+				/>
+				{
+					this.getSetupTasks().map( this.renderSetupTask )
```
Since this is a one liner, I think we should have it on the same line as the brackets.

allendav replied:
Agreed! Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/032e34b77886a58b7e5172370e876738a9c9176d

justinshreve commented on client/extensions/woocommerce/app/dashboard/index.js:
```diff
@@ -3,30 +3,67 @@
-					<p>This is the start of something great!</p>
-					<p>This will be the home for your WooCommerce Store integration with WordPress.com.</p>
-				</Card>
+				{
```
I think this could be condensed to one line (brackets and statement) since it is a one liner.

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

justinshreve commented:
Tested and confirmed all your testing steps work.

For the customizer link: I'm thinking that any time we send someone to `wp-admin` to do something, we should open that in a new tab/window.

jameskoster commented:
Styled!

Wondering if on mobile we should just wrap the buttons beneath the main label or not.

![egg](https://cldup.com/ie3vaY908L-3000x3000.png)

![mobile-egg](https://cldup.com/_giSRCnk1e-3000x3000.png)

jameskoster commented:
> Wondering if on mobile we should just wrap the buttons beneath the main label or not.

Decided to go ahead and do just that.

<img src="https://cldup.com/PPixzBRmse-3000x3000.png" width="200" />

kellychoffman commented:
@jameskoster - Looking good! Just going against the screenshot you posted - should the list go full width on mobile? (ie No space in between the horizontal rule and the edge of the page.) 

Why did you go with non primary compact buttons? I was thinking they would be full size and primary when not checked off (Set up) and then non-primary when checked off (Manage). This would make the to-do's stand out more. Should 'I'm finished setting up' have a bigger button than the tasks above it?

Sidenote: I keep getting an error when I check out the branch, otherwise I'd try a few of these out myself to see how they looked. Error: `Uncaught (in promise) TypeError: Cannot read property 'slug' of null`

jameskoster commented:
> should the list go full width on mobile? (ie No space in between the horizontal rule and the edge of the page.)

I'll try it.

Good point re the buttons, I'll see if I can figure out how to do that xD

jameskoster commented:
Switched to primary buttons, I think this is better;

<img src="https://cldup.com/oh2psy2xJT.png" width="150" />

Couldn't figure out how to make the buttons non-primary when the step has been completed, I'll leave that to someone who understands React xD 

On that note, the import button should be non-primary as well.

I'm also thinking that when all steps are complete the "I'm finished..." button could switch to be primary.

belcherj commented on client/extensions/woocommerce/app/dashboard/index.js:
```diff
@@ -3,30 +3,67 @@
+	renderStoreManagement = () => {
+		return (
+			<Card>
+				<p>This is the start of something great!</p>
```
Desn't hold up the commit but this should be translated even if it is placeholder text

allendav replied:
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/e14d133c34678fb404ae593d8f802590f4c2b3d9

belcherj commented:
If Justin's fixes are made I am ready to Approve.  Ping me on slack once you have done so and let's get this puppy merged.

jameskoster commented:
These items are still outstanding, dunno if you want to do them in a separate PR...

* Make the task buttons non-primary once the task is complete
* Make the import button non-primary
* When the steps are complete, make the "I'm finished..." button primary
* Fix the width attribute on the image

allendav commented:
@jameskoster wrote:

> These items are still outstanding...

tackling these and @belcherj feedback now

allendav commented:
@justinshreve wrote:

> For the customizer link: I'm thinking that any time we send someone to wp-admin to do something, we should open that in a new tab/window.

Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/f468bba8f6d998892a7ef90c5457693da0d70591

allendav commented on client/extensions/woocommerce/app/dashboard/index.js:
```diff
@@ -3,30 +3,67 @@
+
+	return {
+		selectedSite: getSelectedSite( state ),
+		storeSetupCompleted: false
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

allendav commented on client/extensions/woocommerce/app/dashboard/setup-task.js:
```diff
@@ -0,0 +1,89 @@
+ * Internal dependencies
+ */
+import Button from 'components/button';
+import Gridicon from 'gridicons';
```
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/66d13147f190def3689b815188fde46da04a36b4

allendav commented:
@justinshreve wrote:

> Another minor note: I noticed an empty dashboard/selectors.js. It doesn't really affect anything, but probably worth dropping from this PR until something goes in it.

Oops! Removed in https://github.com/Automattic/wp-calypso/pull/14202/commits/b18c751543ceebab4b91ce34526f97b53d6f695e

allendav commented:
@jameskoster wrote:

> Make the task buttons non-primary once the task is complete
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/d3f760dcac1ffae5b7f9f3301198dcc7ed139d3a

> Make the import button non-primary
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/303cfb003dc0caa35b291b8e4e1842055edb4f5e

> When the steps are complete, make the "I'm finished..." button primary
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/acc5cfe7f202714c5e3d6cca505359f162c190fa

> Fix the width attribute on the image
Fixed in https://github.com/Automattic/wp-calypso/pull/14202/commits/877263fee86aa0285e91a13139ee446038cf3afc

allendav commented:
@belcherj @justinshreve @jameskoster - all requests are in - good to go?

kellychoffman commented:
This looks so good! :shipit:

jameskoster commented:
Egg-cellent!

belcherj commented:
I fixed a css merge conflict, LGTM but now I cannot review. :(
